packages/core/src/module-resolver.ts

```diff
@@ -542,12 +542,22 @@ export class Resolver {
   }
 
   private *componentJSCandidates(inPackageName: string) {
+    const extensions = ['.ts', '.gjs', '.gts'];
     yield { prefix: '/components/', suffix: '' };
     yield { prefix: '/components/', suffix: '/component' };
 
+    for (const ext of extensions) {
+      yield { prefix: '/components/', suffix: ext };
+      yield { prefix: '/components/', suffix: `index${ext}` };
```

Comment:
```suggestion
      yield { prefix: '/components/', suffix: `/index${ext}` };
```
if component has a leading slash, should index as well?